helix-view/src/theme.rs

```diff
@@ -297,6 +297,11 @@ impl Theme {
         self.highlights[index]
     }
 
+    #[inline]
+    pub fn scope(&self, index: usize) -> String {
+        self.scopes[index].clone()
```

Comment:
```suggestion
    pub fn scope(&self, index: usize) -> &str {
        &self.scopes[index]
```

it's usually a nicer API to return borrowed data (`&str`) rather than owned (`String`) so that the caller can decide when/if to own the data. For example a caller of this function might only need to check `scope.starts_with("keyword")` which it can do without allocating a `String`.

Callers that need to allocate though (like `tree_sitter_highlight_name`) can then make the call themselves:

```rust
let content = cx.editor.theme.scope(highlight.0).to_string();
```

Reply:
That makes sense. Appreciate the explanation. Addressed in most recent commit.